addons/zylann.hterrain/tools/brush/brush_editor.gd

```diff
@@ -190,13 +192,25 @@ func set_display_mode(mode: int):
 func _on_size_slider_value_changed(v: float):
 	if _terrain_painter != null:
 		_terrain_painter.set_brush_size(int(v))
-	_size_value_label.text = str(v)
+	_size_spin_box.set_value_no_signal(v)
+
+
+func _on_size_spin_box_value_changed(v: float):
+	if _terrain_painter != null:
+		_terrain_painter.set_brush_size(int(v))
+	_size_slider.set_value_no_signal(v)
 
 
 func _on_opacity_slider_value_changed(v: float):
 	if _terrain_painter != null:
 		_terrain_painter.set_opacity(_opacity_slider.ratio)
-	_opacity_value_label.text = str(v)
+	_opacity_spin_box.set_value_no_signal(v)
+
+
+func _on_opacity_spin_box_value_changed(v: float):
+	if _terrain_painter != null:
+		_terrain_painter.set_opacity(_opacity_slider.ratio)
+	_opacity_slider.set_value_no_signal(v)
```

Comment:
Actually I think that line should come first in the function, because `_terrain_painter.set_opacity(_opacity_slider.ratio)`

Reply:
Yeah, you are correct. Just updated it.